src/Symfony/Component/Form/Tests/Extension/Core/Type/MoneyTypeTest.php

```diff
@@ -123,4 +123,23 @@ public function testHtml5EnablesSpecificFormatting()
         $this->assertSame('12345.60', $form->createView()->vars['value']);
         $this->assertSame('number', $form->createView()->vars['type']);
     }
+
+    public function testHtml5EnablesSpecificFormattingWithIntegerFormat()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, ['html5' => true, 'scale' => 2, 'input' => 'integer']);
```

Comment:
I am not sure that a scale option with a value other than `0` plays well with how the `input` option is implemented currently.

But, this makes me wonder if the way this option is currently implemented actually makes sense. So maybe we should indeed allow the `division` to be greater than 1 and take it into account to decide whether we want to render a `number` or an `integer` widget.